iceoryx_posh/source/roudi/process_manager.cpp

```diff
@@ -533,6 +533,101 @@ void ProcessManager::addPublisherForProcess(const RuntimeName_t& name,
         [&]() { LogWarn() << "Unknown application " << name << " requested a PublisherPort."; });
 }
 
+
+void ProcessManager::addClientForProcess(const RuntimeName_t& name,
+                                         const capro::ServiceDescription& service,
+                                         const popo::ClientOptions& clientOptions,
+                                         const PortConfigInfo& portConfigInfo) noexcept
+{
+    searchForProcessAndThen(
```

Comment:
Side note which makes me wondering. Shouldn't `searchForProcessAndThen` return an `cxx::optional`? Then one can write `searchForProcess(name).and_then([&](Process & process){ ...`

I know it is not the task of this PR but this really bugs me. What do you think, could you make a senior happy by adjusting this?

Reply:
Sure

Reply:
Done. Happy weekend :)